### PR TITLE
Tolerations for drain detection canaries

### LINKED ISSUE #3813: Ceph: drain-detection logic should take tolerations from the placement configuration.
Currently managedPDBs won't work with tolerations. Will have a fix tomorrow.

### PR DESCRIPTION
**Pull request by rohantmp**
Should be ready, need to verify behavior by testing.

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #3813 

**Checklist:**

- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [x] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [x] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph min]

### CODE REVIEW

**Comment by rohantmp** [comment]
will add a unit test.

**Comment by rohantmp** [comment]
done

**Comment by rohantmp** [comment]
@leseb

File: pkg/operator/ceph/disruption/nodedrain/reconcile.go [hunk 1]
```diff
@@ -88,11 +88,26 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
...
 	occupiedByOSD := false
 	for _, osdPod := range osdPodList.Items {
 		if osdPod.Spec.NodeName == request.Name {
 			occupiedByOSD = true
-			break
+			// get the osd tolerations
+			for _, osdToleration := range osdPod.Spec.Tolerations {
+				if osdToleration.Key == "node.kubernetes.io/unschedulable" {
+					labels := osdPod.GetLabels()
+					logger.Errorf(
```
Comment (leseb) [inline pkg/operator/ceph/disruption/nodedrain/reconcile.go]: So we just log the error and that's it? Don't we want to return here?

Comment (rohantmp) [reply pkg/operator/ceph/disruption/nodedrain/reconcile.go]: This just means that this particular OSD won't get drained at all as it tolerates the drain. We log error as this is bad practice (a drain might precede some other disruption), but there's not much else we can do. It doesn't effect the function of the drain canaries.

Comment (leseb) [reply pkg/operator/ceph/disruption/nodedrain/reconcile.go]: Thanks!

**Comment by rohantmp** [comment]
Added the commit message.

**Comment by leseb** [comment]
Looks like it's not passing CI.

**Comment by travisn** [comment]
@rohantmp want to open a backport PR?

**Pull request merged**